src/kaltura-player.js

```diff
@@ -75,6 +74,7 @@ class KalturaPlayer extends FakeEventTarget {
     Object.values(CoreEventType).forEach(coreEvent => this._eventManager.listen(this._localPlayer, coreEvent, e => this.dispatchEvent(e)));
     this._addBindings();
     this._playlistManager.configure(options.playlist);
+    this.configure({plugins});
```

Comment:
The `UIWrapper` uses the `player.uiComponents` which filled by the plugins while their initialization. So we can't move it after the `UIWrapper` initialization.

Reply:
So the correct flow here is to actually fix the UI manager issue.
@Yuvalke the correct expectation is that if someone calls reset on player it will actually reset the UI as well, even if it is in transit.
Issue is that as we call reset on error event and event didn't yet reached UI we can't really stop it - chicken and egg kind of situation.
I suggest you and @yairans discuss this so @yairans can have more context and let's try to figure it out.

Reply:
@yairans @OrenMe we use pass on configure with setConfig and pass the uiComponents there instead, I'll need to verify it with plugins like this

Reply:
@Yuvalke plugins expose UI components, it is not just app so need them to be available before.
But now with dynamic injection this is not true any more and I assume we can find a solution where UI is configured and then plugins and then we register plugins UI components on UI.
@yairans FYI

Reply:
https://github.com/kaltura/playkit-js-ui/pull/563 - implement the add UI component which provides an option to fix it

Reply:
@yairans updated as we discuss, external UI components will update via config and plugins will update via API.